perllib/FixMyStreet/App/Controller/Admin.pm

```diff
@@ -1856,6 +1858,43 @@ sub fetch_body_areas : Private {
     $c->stash->{fetched_areas_body_id} = $body->id;
 }
 
+sub update_extra_fields : Private {
+    my ($self, $c, $object) = @_;
+
+    my @indices = grep { /^metadata\[\d+\]\.code/ } keys %{ $c->req->params };
+    @indices = sort map { $_ =~ /(\d+)/ } @indices;
+
+    $object->set_extra_fields(());
+    foreach my $i (@indices) {
+        my $meta = {};
+        $meta->{code} = $c->get_param("metadata[$i].code");
+        next unless $meta->{code};
+        $meta->{order} = int $c->get_param("metadata[$i].order");
+        $meta->{datatype} = $c->get_param("metadata[$i].datatype");
+        $meta->{required} = $c->get_param("metadata[$i].required") eq 'on' ? 'true' : 'false';
+        $meta->{variable} = $c->get_param("metadata[$i].variable") eq 'on' ? 'true' : 'false';
+        $meta->{description} = $c->get_param("metadata[$i].description");
+        $meta->{datatype_description} = $c->get_param("metadata[$i].datatype_description");
+
+        if ( $meta->{datatype} eq "singlevaluelist" ) {
+            $meta->{values} = [];
+            my $re = "^metadata\\[$i\\]\\.values\\[\\d+\\]\\.key";
```

Comment:
If you wrap $re in `qr{}` so it's compiled immediately, then you can `s/\\\\/\\/` and make this look slightly nicer ;-)